tests/integration/golang/mlflow/artifact/get_artifact_s3_test.go

```diff
@@ -0,0 +1,269 @@
+				LifecycleStage: models.LifecycleStageActive,
+			})
+			assert.Nil(s.T(), err)
+
+			// upload artifact root object to S3
+			putObjReq := &s3.PutObjectInput{
+				Key:    aws.String(fmt.Sprintf("1/%s/artifacts/artifact.file", runID)),
+				Body:   strings.NewReader(`content`),
+				Bucket: aws.String(tt.bucket),
+			}
+			_, err = s.s3Client.PutObject(context.Background(), putObjReq)
+			assert.Nil(s.T(), err)
+
+			// upload artifact subdir object to S3
+			putObjReq = &s3.PutObjectInput{
+				Key: aws.String(fmt.Sprintf(
+					"1/%s/artifacts/artifact.subdir/artifact.file",
+					runID),
+				),
+				Body:   strings.NewReader(`subdir-object-content`),
+				Bucket: aws.String(tt.bucket),
+			}
+			_, err = s.s3Client.PutObject(context.Background(), putObjReq)
+			assert.Nil(s.T(), err)
+
+			// make API call for root object
+			query, err := urlquery.Marshal(request.GetArtifactRequest{
+				RunID: run.ID,
+				Path:  "artifact.file",
+			})
+			assert.Nil(s.T(), err)
+
+			resp, err := s.serviceClient.DoStreamRequest(
+				http.MethodGet,
+				fmt.Sprintf("%s%s?%s", mlflow.ArtifactsRoutePrefix, mlflow.ArtifactsGetRoute, query),
+				nil,
+			)
+			assert.Nil(s.T(), err)
+			assert.Equal(s.T(), "content", string(resp))
+
+			// make API call for subdir object
+			query, err = urlquery.Marshal(request.GetArtifactRequest{
+				RunID: run.ID,
+				Path:  "artifact.subdir/artifact.file",
+			})
+			assert.Nil(s.T(), err)
+
+			resp, err = s.serviceClient.DoStreamRequest(
+				http.MethodGet,
+				fmt.Sprintf("%s%s?%s", mlflow.ArtifactsRoutePrefix, mlflow.ArtifactsGetRoute, query),
+				nil,
+			)
+			assert.Nil(s.T(), err)
+			assert.Equal(s.T(), "subdir-object-content", string(resp))
+		})
+	}
+}
+
+func (s *GetArtifactS3TestSuite) Test_Error() {
+	defer func() {
+		assert.Nil(s.T(), s.experimentFixtures.UnloadFixtures())
+	}()
+
+	// create test experiment
+	experiment, err := s.experimentFixtures.CreateExperiment(context.Background(), &models.Experiment{
+		Name:             "Test Experiment In Bucket bucket1",
+		LifecycleStage:   models.LifecycleStageActive,
+		ArtifactLocation: "s3://bucket1/1",
+	})
+	assert.Nil(s.T(), err)
+
+	// create test run
+	runID := strings.ReplaceAll(uuid.New().String(), "-", "")
+	_, err = s.runFixtures.CreateRun(context.Background(), &models.Run{
+		ID:             runID,
+		Status:         models.StatusRunning,
+		SourceType:     "JOB",
+		ExperimentID:   *experiment.ID,
+		ArtifactURI:    fmt.Sprintf("%s/%s/artifacts", experiment.ArtifactLocation, runID),
+		LifecycleStage: models.LifecycleStageActive,
+	})
+	assert.Nil(s.T(), err)
+
+	// upload artifact subdir object to S3
+	putObjReq := &s3.PutObjectInput{
+		Key:    aws.String(fmt.Sprintf("1/%s/artifacts/artifact.subdir/artifact.file", runID)),
+		Body:   strings.NewReader(`content`),
+		Bucket: aws.String("bucket1"),
+	}
+	_, err = s.s3Client.PutObject(context.Background(), putObjReq)
+	assert.Nil(s.T(), err)
+
+	testData := []struct {
```

Comment:
sometimes we use `testCases` sometimes `testData`, can you please unity it? for me `tests` just works really well.

Reply:
sure